### PR TITLE
vim-patch: bump VIM_VERSION from 8.0 => 8.1

### PR DESCRIPTION
**Pull request by justinmk**
There are 6 remaining 8.0.x patches, tracked in:
https://github.com/neovim/neovim/issues/5431

### CODE REVIEW

File: scripts/vim-patch.sh [hunk 1]
```diff
@@ -488,7 +488,7 @@ list_vimpatch_tokens() {
 list_vimpatch_numbers() {
   # Transform "vim-patch:X.Y.ZZZZ" to "ZZZZ".
   list_vimpatch_tokens | while read -r vimpatch_token; do
-    echo "$vimpatch_token" | grep '8\.0\.' | sed 's/.*vim-patch:8\.0\.\([0-9a-z]\+\).*/\1/'
+    echo "$vimpatch_token" | grep '8\.1\.' | sed -E 's/.*vim-patch:8\.1\.([0-9a-z]+).*/\1/'
```
Comment (justinmk) [inline scripts/vim-patch.sh:491]: `-E` allows `vim-patch.sh` to work on macOS `sed`.  And `-E` is already used elsewhere in this script.

File: scripts/vimpatch.lua [hunk 2]
```diff
@@ -1,7 +1,7 @@
 --  Updates version.c list of applied Vim patches.
 --
 --  Usage:
---    VIM_SOURCE_DIR=~/neovim/.vim-src/ nvim -i NONE -u NONE --headless +'luafile ./scripts/vimpatch.lua' +q
+--    VIM_SOURCE_DIR=~/neovim/.vim-src/ nvim -V1 -es +'luafile ./scripts/vimpatch.lua' +q
```
Comment (jamessan) [inline scripts/vimpatch.lua:4]:
```suggestion
--    VIM_SOURCE_DIR=~/neovim/.vim-src/ nvim -V1 -i NONE -es +'luafile ./scripts/vimpatch.lua' +q
```

**Review by jamessan — APPROVED** [review]

File: scripts/vimpatch.lua (hunk 2 above)
Comment (justinmk) [reply scripts/vimpatch.lua:4]: gah, we should change `-es` to do that by default.

Comment (jamessan) [reply scripts/vimpatch.lua]: We specifically document that it doesn't do that. :⁠-⁠)

Comment (justinmk) [reply scripts/vimpatch.lua]: yes, but we should change it

**Pull request merged**